File: calculator.py
def add(a, b):
    return a + b


def subtract(a, b):
    return a - b


def multiply(a, b):
    return a * b


def divide(a, b):
    if b == 0:
        return "Error: Division by zero is not allowed."
    return a / b


operations_dict = {
    "+": add,
    "-": subtract,
    "*": multiply,
    "/": divide
}

number1 = int(input("Enter 1st number: "))

for symbol in operations_dict:
    print(symbol)

continue_flag = True
while continue_flag:
    op_symbol = input("Pick an Operation: ")

    if op_symbol not in operations_dict:
        print("Invalid operation. Please try again.")
        continue

    number2 = int(input("Enter 2nd number: "))

    calculator_function = operations_dict[op_symbol]
    output = calculator_function(number1, number2)

    print(f"{number1} {op_symbol} {number2} = {output}")

    should_continue = input(f"Enter 'y' to continue calculation with {output} or 'n' to exit: ").lower()

    if should_continue == 'y':
        number1 = output
    else:
        continue_flag = False
        print("Bye")

File: calculator.cpp
#include<iostream>
using namespace std;
int main()
{
    int choice,number1,number2;
    int addition,subtraction,multiplication,division,modulus;
    do{
    cout<<"************************"<<endl;
    cout<<"1. Addition"<<"\n"<<"2. Subtraction"<<"\n"<<"3. Multiplication"<<"\n"<<"4. Division"<<"\n"<<"5. Modulus"<<"\n"<<"6. Exit"<<endl;
    cout<<"Enter your choice: "<<endl;
    cin>>choice;


    switch(choice)
    {
        case 1:
        {
                cout<<"You chose 1.....Addition"<<endl;
                cout<<"Enter 2 numbers seperated by a space: "<<endl;
                cin>>number1>>number2;
                addition=number1+number2;
                cout<<"Addition is:"<<addition<<endl;
                break;
        }
        case 2:
        {
                cout<<"You chose 2.....Subtraction"<<endl;
                cout<<"Enter 2 numbers seperated by a space: "<<endl;
                cin>>number1>>number2;
                subtraction=number1-number2;
                cout<<"Subtraction is:"<<subtraction<<endl;
                break;
        }
        case 3:
        {
                cout<<"You chose 3.....Multiplication"<<endl;
                cout<<"Enter 2 numbers seperated by a space: "<<endl;
                cin>>number1>>number2;
                multiplication=number1*number2;
                cout<<"Multiplication is:"<<multiplication<<endl;
                break;
        }
        case 4:
        {
                cout<<"You chose 4.....Division"<<endl;
                cout<<"Enter 2 numbers seperated by a space: "<<endl;
                cin>>number1>>number2;
                division=number1/number2;
                cout<<"Division is:"<<division<<endl;
                break;
        }
        case 5:
        {
                cout<<"You chose 5.....Modulus"<<endl;
                cout<<"Enter 2 numbers seperated by a space: "<<endl;
                cin>>number1>>number2;
                modulus=number1%number2;
                cout<<"Modulus is:"<<modulus<<endl;
                break;
        }
        case 6:
        {
            cout<<"Good bye !"<<endl;
            break;
        }
        default:
        {
            cout<<"please select valid choice !"<<endl;
            break;
        
    }}}while(choice!=6);
    return 0;
}
